src/main/java/sparksoniq/jsoniq/item/ArrayItem.java

```diff
@@ -73,9 +73,7 @@ public int getSize() {
 
     @Override
     public boolean isTypeOf(ItemType type) {
-        if (type.getType().equals(ItemTypes.ArrayItem) || super.isTypeOf(type))
-            return true;
-        return false;
+        return type.getType().equals(ItemTypes.ArrayItem) || super.isTypeOf(type);
```

Comment:
Very good. To take it to the next level in terms of code readability, you could also avoid the super.isTypeOf recursive call and "flatten out" the tests in each class.

But this is mostly cosmetic as the super call will anyway probably be inlined by the optimizer. So I can live with the current implementation, which does the job.

Reply:
I think I would prefer to leave the implementation that relies on inheritance from the defined type hierarchy, rather than explicitly flattening out the different super types. I think it is cleaner and easier to understand the concept of what is being checked. Open to discussion though.